### PR TITLE
Bucket type to bucket hash

### PR DESCRIPTION
**Pull request by bhollis**
I noodled on this a while ago - trying to replace as much reference to the DIM-specific "type" string as possible with bucket hashes. I couldn't quite get them all, but I think it's still better this way?

### CODE REVIEW

File: src/app/inventory/DraggableInventoryItem.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ export default function DraggableInventoryItem({ children, item }: Props) {
       type: item.location.inPostmaster
         ? 'postmaster'
         : item.notransfer
-        ? `${item.owner}-${item.bucket.type}`
-        : item.bucket.type!,
+        ? `${item.owner}-${item.bucket.hash}`
+        : item.bucket.hash.toString(),
```
Comment (bhollis) [inline src/app/inventory/DraggableInventoryItem.tsx:22]: I think there are a few places where I missed the implication of this change.

File: src/app/shell/filters.ts [hunk 2]
```diff
@@ -99,9 +100,9 @@ const D1_MATERIAL_SORT_ORDER = [
 
 // Bucket IDs that'll never be sorted.
 const ITEM_SORT_DENYLIST = new Set([
-  2197472680, // Bounties (D1)
-  375726501, // Mission (D1)
-  1801258597, // Quests (D1)
+  D1BucketHashes.Bounties, // Bounties (D1)
+  BucketHashes.Engrams, // Mission (D1)
```
Comment (delphiactual) [inline src/app/shell/filters.ts:104]:
```suggestion
  D1BucketHashes.Missions,
```

File: src/app/destiny1/d1-buckets.ts [hunk 3]
```diff
@@ -11,34 +13,34 @@ import type { D1ManifestDefinitions } from './d1-definitions';
...
-  3551918588: 'Gauntlets',
-  3796357825: 'Horn',
-  3865314626: 'Material',
-  4023194814: 'Ghost',
-  4274335291: 'Emblem',
+  [BucketHashes.ChestArmor]: 'Chest',
+  [BucketHashes.LegArmor]: 'Leg',
+  [BucketHashes.LostItems]: 'LostItems',
+  [BucketHashes.Ships]: 'Ship',
+  [BucketHashes.Engrams]: 'Missions', // Yes, engrams (D1) and missions (D1) have the same hash
```
Comment (delphiactual) [inline src/app/destiny1/d1-buckets.ts:20]:
```suggestion
  [D1BucketHashes.Missions]: 'Missions', // Yes, engrams (D1) and missions (D1) have the same hash
```

File: src/app/shell/filters.ts [hunk 4]
```diff
@@ -99,9 +100,9 @@ const D1_MATERIAL_SORT_ORDER = [
...
+  D1BucketHashes.Quests, // Quests (D1)
```
Comment (delphiactual) [inline src/app/shell/filters.ts:105]:
```suggestion
  D1BucketHashes.Bounties,
  D1BucketHashes.Missions,
  D1BucketHashes.Quests,

File: src/app/destiny2/d2-buckets.ts [hunk 5]
```diff
@@ -12,39 +13,39 @@ import { D2ManifestDefinitions } from './d2-definitions';
...
+  [BucketHashes.Engrams]: 'Engrams',
+  [BucketHashes.PowerWeapons]: 'Power',
+  [BucketHashes.Auras]: 'Auras',
+  [BucketHashes.SpecialOrders]: 'SpecialOrders',
+  [BucketHashes.KineticWeapons]: 'KineticSlot',
+  [BucketHashes.ClassArmor]: 'ClassItem',
+  [BucketHashes.Vehicle]: 'Vehicle',
+  [BucketHashes.Consumables]: 'Consumables',
+  [BucketHashes.General]: 'General',
+  [BucketHashes.Emotes_Invisible]: 'Emotes',
```
Comment (nev-r) [inline src/app/destiny2/d2-buckets.ts:45]: pretty sure this should be Emotes_Equippable (individual emotes)
not Emotes_Invisible (this button)
14aa589109bbd9e95df0757b23511b73 1

Comment (nev-r) [reply src/app/destiny2/d2-buckets.ts:45]: but it's unclear if this is actually used anywhere, aside from a yet-unrealized ability to add emotes to loadouts

**Comment by bhollis** [comment]
OK, fixed this up a bit. Would appreciate a test pass.

**Comment by nev-r** [comment]
tried to click the major stuff, all tabs, dragging, compare, loadouts, organizer, optimizer, etc etc etc. but buckets are pervasive so i feel like i can't guarantee no surprises. looks good though. sooner it's merged the safer

**Review by nev-r — APPROVED** [review]

**Pull request merged**